src/libcore/ptr/mod.rs

```diff
@@ -112,11 +112,17 @@ mod mut_ptr;
 ///   as the compiler doesn't need to prove that it's sound to elide the
 ///   copy.
 ///
+/// * It can be used to drop [pinned] data (pinned data must not be moved
+///   before it is dropped).
```

Comment:
```suggestion
///   before it is dropped), except if `T` is `repr(packed).
```
I feel currently this reads a bit like a contradiction, where it says "can be used to drop pinned data (because in-place)" but later says "but fields of packed structs are not dropped in-place".

Reply:
AIUI, packed structs with destructors cannot be pinned to begin with, and packed structs without destructors are presumably safe to drop using this function anyway?

Reply:
They cannot be pinned *precisely because of what this comment says*. If someone tries anyway, this is the place where they are violating the rules. So this is the place where that should be mentioned.

The compiler won't stop you when you try to pin something packed.

Reply:
Fixed. I put the caveat before the part in brackets because I felt it was awkward to read otherwise.